test/integration/install/install_uninstall_test.go

```diff
@@ -0,0 +1,217 @@
+
+	"github.com/shirou/gopsutil/process"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/compose"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	osReleasePath  = "/etc/os-release"
+	maxFileSize    = int64(20000000)
+	maxInstallTime = 30 * time.Second
+)
+
+var (
+	AGENT_PACKAGE_FILENAME = os.Getenv("PACKAGE_NAME")
+	agentContainer         *testcontainers.DockerContainer
+)
+
+func setupTestContainer(t *testing.T) {
+	comp, err := compose.NewDockerCompose("docker-compose.yml")
+	assert.NoError(t, err, "NewDockerComposeAPI()")
+
+	t.Cleanup(func() {
+		assert.NoError(t, comp.Down(context.Background(), compose.RemoveOrphans(true), compose.RemoveImagesLocal), "compose.Down()")
+	})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	require.NoError(t, comp.WaitForService("agent", wait.ForHTTP("/")).WithEnv(
+		map[string]string{
+			"PACKAGE_NAME": os.Getenv("PACKAGE_NAME"),
+			"BASE_IMAGE":   os.Getenv("BASE_IMAGE"),
+		},
+	).Up(ctx, compose.Wait(true)), "compose.Up()")
+
+	agentContainer, err = comp.ServiceContainer(context.Background(), "agent")
+	require.NoError(t, err)
+}
+
+// TestAgentManualInstallUninstall tests Agent Install and Uninstall.
+// Verifies that agent installs with correct output and files.
+// Verifies that agent uninstalls and removes all the files.
+func TestAgentManualInstallUninstall(t *testing.T) {
+	// Check the environment variable $PACKAGE_NAME is set
+	require.NotEmpty(t, AGENT_PACKAGE_FILENAME, "Environment variable $PACKAGE_NAME not set")
+
+	setupTestContainer(t)
+
+	exitCode, o, err := agentContainer.Exec(context.Background(), []string{"cat", osReleasePath})
+	assert.NoError(t, err)
+	osReleaseContent, err := io.ReadAll(o)
+	assert.NoError(t, err)
+	assert.Equal(t, 0, exitCode)
+	assert.NotEmpty(t, osReleaseContent, "os release file empty")
+
+	expectedInstallLogMsgs := map[string]string{
+		"InstallFoundNginxAgent": "Found nginx-agent /usr/bin/nginx-agent",
+		"InstallAgentToRunAs":    "nginx-agent will be configured to run as same user",
+		"InstallAgentSuccess":    "NGINX Agent package has been successfully installed.",
+		"InstallAgentStartCmd":   "sudo systemctl start nginx-agent",
+	}
+
+	expectedUninstallLogMsgs := map[string]string{
+		"UninstallAgent":             "Removing nginx-agent",
+		"UninstallAgentPurgingFiles": "Purging configuration files for nginx-agent",
+	}
+
+	expectedAgentPaths := map[string]string{
+		"AgentConfigFile":        "/etc/nginx-agent/nginx-agent.conf",
+		"AgentDynamicConfigFile": "/etc/nginx-agent/agent-dynamic.conf",
+	}
+
+	agentPackageFilePath := getPackagePath(string(osReleaseContent))
+
+	// Check the agent package is present
+	agentPkg, err := agentContainer.CopyFileFromContainer(context.Background(), agentPackageFilePath)
+	assert.NoError(t, err)
+
+	a, err := io.ReadAll(agentPkg)
```

Comment:
Can you add a more descriptive variable name? Something that helps the code reader with the context